### DISCUSSION TITLE
Which voltage divider for 3.3v board?

### DISCUSSION THREAD

**Discussion by fyrus7** [rlogiacco/BatterySense#48] [Q&A] [answered]
The main readme suggest voltage divider like this for 3.3v board

```
Battery + ----
             |
            R1 (4.7kΩ)
             |
             |--------- A0
             |
            R2 (10kΩ)
             |
Battery - ----
```

How to connect this actually? Is it like this?

```
Battery + ---------- VCC pin
                     |
                    R1 (4.7kΩ)
                     |
                     |--------- A0 pin
                     |
                    R2 (10kΩ)
                     |
Battery - ---------- GND pin
```

**Comment by rlogiacco** [comment]
No, the positive side of the battery should go into **Vin** (to be further regulated to 3.3V), the negative into GND.
If your board doesn't have an onboard regulator you should buy or build one: a battery cannot provide a stable input voltage.

**Comment by fyrus7** [comment]
> No, the positive side of the battery should go into **Vin** (to be further regulated to 3.3V), the negative into GND. If your board doesn't have an onboard regulator you should buy or build one: a battery cannot provide a stable input voltage.

So the divider is from VCC and GND > junction to A0?

**Comment by rlogiacco** [comment, marked as answer]
First of all, I migrated this from issues to discussions: that's the more appropriate place for a question.

Second, you never stated what is your battery configuration, specifically the maximum voltage. I will here assume, looking at your resistor values, that are dealing with a 3 AA battery pack (`4.5V` max) or a single LiIon cell (`4.2V` max), but please do not proceed if my assumption is wrong.

Third, usually, a board has a `VCC` pin and a `Vin` pin: the difference is the second goes through a voltage regulator and can accept higher voltage than `3.3V` (or whatever is the board voltage). If your board doesn't have a `Vin` pin you are probably dealing with a board without a voltage regulator, that means you cannot power the board directly from a battery because a battery is not a regulated voltage source: you need to add a voltage regulator between the battery and the board. 

**DO NOT** use `VCC` to power your board from a battery, you risk burning the microcontroller.

[Here is the circuit](https://www.falstad.com/circuit/circuitjs.html?ctz=CQAgjCAMB0l3BWcMBMcUHYMGZIA4UA2ATmIxAUgpABZsKBTAWjDACgAnETFEbNbhl4oUNKMnhsAbuGK9+VMHPAo84qjQ3RyVXdASdBvMKqPdR4mhjhsA7rPkCefAZDtmFfbIRdUU7pWNTZxM1dnsQ02xvFTV-CKFzMWifETF-NHJA2PBCRVMxACEAQQAVAB0ABzZMh19cqk8isqYayHIUnO8wgpAANQBLADs2jpjPbvqxAHEAOQARUa9Ui0m02hBitwB7cUIxDUhSahhIJD9xbDZsNSoAMQhFZAgWEAAlBgBnAc+AFwBDIYAYwYbCAA) I'm proposing, where `Vin` and `B+` are the same voltage, `GND` and `B-` are the same voltage and the voltage divider goes between those two nodes: the goal is to provide `A0` a compatible voltage for a `3.3V` board.

Please mark this question as answered if this sounds satisfactory to you.

**Reply by fyrus7** [reply]
Yes you are correct!

Battery : Liion 3.7v  (4.2v max)
MCU     : Arduino Pro Mini 3.3v

For anyone who looking to do this with Pro Mini 3.3v powered by 3.7v Liion

Battery + ---- Vin
&emsp;  &emsp;           |
&emsp;  &emsp;          R1 (4.7kΩ)
&emsp;  &emsp;           |
&emsp;  &emsp;           |--------- A0
&emsp;  &emsp;           |
&emsp;  &emsp;          R2 (10kΩ)
&emsp;  &emsp;           |
Battery - ---- GND
